### PR TITLE
Add step to upload artifact with contracts' json files

### PR DESCRIPTION
**Pull request by mikogs**

### CODE REVIEW

File: .github/workflows/deploy-the-button.yaml [hunk 1]
```diff
@@ -143,6 +143,30 @@ jobs:
           source contracts/env/${{ env.CONTRACTS_ENVFILE }} && echo -n "$NODE" > env_NODE.txt
           aws s3 cp env_NODE.txt s3://${{ secrets.CI_S3BUCKET_NAME }}/${{ env.S3BUCKET_PATH }}/env_NODE.txt
 
+      - name: Rename metadata and addresses JSON files to artifact upload
+        shell: bash
+        run: |
+          cp contracts/addresses.json addresses.featurenet.json
```
Comment (fbielejec) [inline .github/workflows/deploy-the-button.yaml:149]: you can use you can use `env.CONTRACTS_ENVFILE` here

File: .github/workflows/deploy-the-button.yaml [hunk 2]
```diff
@@ -143,6 +143,30 @@ jobs:
...
+          for i in ticket_token marketplace button game_token access_control; do \
+            cp contracts/"$i"/target/ink/metadata.json metadata_"$i".featurenet.json \
```
Comment (fbielejec) [inline .github/workflows/deploy-the-button.yaml:151]: same here

File: .github/workflows/deploy-the-button.yaml [hunk 3]
```diff
@@ -143,6 +143,30 @@ jobs:
...
+          done
+
+      - name: Upload artifact with contract addresses
+        uses: actions/upload-artifact@v3
+        with:
+          name: contracts-addresses
+          path: addresses.featurenet.json
```
Comment (fbielejec) [inline .github/workflows/deploy-the-button.yaml:158]: use env var

File: .github/workflows/deploy-the-button.yaml [hunk 4]
```diff
@@ -143,6 +143,30 @@ jobs:
...
+          if-no-files-found: error
+          retention-days: 90
+
+      - name: Upload artifact with contract metadata
+        uses: actions/upload-artifact@v3
+        with:
+          name: contracts-metadata
+          path: metadata_*.featurenet.json
```
Comment (fbielejec) [inline .github/workflows/deploy-the-button.yaml:166]: same

File: .github/workflows/deploy-the-button.yaml (hunk 1 above)
Comment (mikogs) [reply .github/workflows/deploy-the-button.yaml]: changed

File: .github/workflows/deploy-the-button.yaml (hunk 2 above)
Comment (mikogs) [reply .github/workflows/deploy-the-button.yaml]: changed

File: .github/workflows/deploy-the-button.yaml (hunk 3 above)
Comment (mikogs) [reply .github/workflows/deploy-the-button.yaml]: changed

File: .github/workflows/deploy-the-button.yaml (hunk 4 above)
Comment (mikogs) [reply .github/workflows/deploy-the-button.yaml]: changed

**Review by Kosiare — APPROVED** [review]

**Review by fbielejec — COMMENTED** [review]
Could we trigger the workflow on:
- push to the benjamin branch with changes done to the contracts/ directory
- it should upload all of the produced artifacts to S3 as well as publish them as GH action artifacts
- it should also trigger a build & deployment of both the fronted and the indexer (they should fetch updated artifacts from S3)

**Pull request merged**